src/ch09-02-recoverable-errors-with-result.md

```diff
@@ -0,0 +1,817 @@
+`Result` using a basic tool: the `match` expression that we discussed in
+Chapter 6.
+
+Devemos fazer com que o código na Lista 9-3 faça diferentes ações dependendo
+do valor retornado por `File::open`. A Lista 9-4 mostra uma maneira de lidar
+com o `Result` usando uma ferramenta básica: a expressão `match` que discutimos
+no Capítulo 6.
+
+<span class="filename">Filename: src/main.rs</span>
+
+```rust,should_panic
+use std::fs::File;
+
+fn main() {
+    let f = File::open("hello.txt");
+
+    let f = match f {
+        Ok(file) => file,
+        Err(error) => {
+            panic!("There was a problem opening the file: {:?}", error)
+        },
+    };
+}
+```
+<span class="filename">Filename: src/main.rs</span>
+
+```rust,should_panic
+use std::fs::File;
+
+fn main() {
+    let f = File::open("hello.txt");
+
+    let f = match f {
+        Ok(file) => file,
+        Err(error) => {
+            panic!("Houve um problema ao abrir o arquivo: {:?}", error)
+        },
+    };
+}
+```
+
+<span class="caption">Listing 9-4: Using a `match` expression to handle the
+`Result` variants we might have</span>
+
+<span class="caption">Lista 9-4: Usando uma expressão `match` para tratar as
+variantes de `Result` que podemos encontrar</span>
+
+Note that, like the `Option` enum, the `Result` enum and its variants have been
+imported in the prelude, so we don’t need to specify `Result::` before the `Ok`
+and `Err` variants in the `match` arms.
+
+Note que, como no enum `Option`, o enum `Result` e suas variantes foram importadas
+no prelúdio, então não precisamos especificar `Result::` antes das variantes `Ok` 
+e `Err` nas linhas de `match`.
+
+Here we tell Rust that when the result is `Ok`, return the inner `file` value
+out of the `Ok` variant, and we then assign that file handle value to the
+variable `f`. After the `match`, we can then use the file handle for reading or
+writing.
+
+Aqui dizemos ao Rust que quando o resultado é `Ok` ele deve retornar o valor
+interno `file` de dentro da variante `Ok` e nós então podemos atribuir este
+valor de *handle* de arquivo à variável `f`. Depois do `match`, nós podemos então
+usar o *handle* de arquivo para ler ou escrever.
+
+The other arm of the `match` handles the case where we get an `Err` value from
+`File::open`. In this example, we’ve chosen to call the `panic!` macro. If
+there’s no file named *hello.txt* in our current directory and we run this
+code, we’ll see the following output from the `panic!` macro:
+
+A outra linha de `match` trata do caso em que recebemos um valor de `Err` de
+`File::open`. Nesse exemplo, nós escolhemos chamar a macro `panic!`. Se não
+há nenhum arquivo chamado *hello.txt* no diretório atual e rodarmos esse código,
+veremos a seguinte saída da macro `panic!`:
+
+```text
+thread 'main' panicked at 'There was a problem opening the file: Error { repr:
+Os { code: 2, message: "No such file or directory" } }', src/main.rs:9:12
+```
```

Comment:
Como a frase `There was a problem opening the file` foi traduzida no código, deve ser traduzida nessa mensagem de saída também.

Reply:
Obrigado! Ainda não consegui tirar um tempo pra começar a aplicar a revisão mas está na minha lista, começo em breve!